src/realm_dart_sync.cpp

```diff
@@ -21,32 +21,29 @@
 #include "realm_dart.hpp"
 #include "realm_dart_sync.h"
 
-RLM_API void realm_dart_http_request_callback(realm_userdata_t userdata, const realm_http_request_t request, void* request_context) {
-    // the pointers in error are to stack values, we need to make copies and move them into the scheduler invocation
+RLM_API void realm_dart_http_request_callback(realm_userdata_t userdata, realm_http_request_t request, void* request_context) {
+    // the pointers in request are to stack values, we need to make copies and move them into the scheduler invocation
     struct request_copy_buf {
         std::string url;
         std::string body;
         std::map<std::string, std::string> headers;
         std::vector<realm_http_header_t> headers_vector;
     } buf;
 
-    realm_http_request_t request_copy = request; // copy struct
-
     buf.url = request.url;
-    request_copy.url = buf.url.c_str();
     buf.body = std::string(request.body, request.body_size);
-    request_copy.body = buf.body.data();
-
     buf.headers_vector.reserve(request.num_headers);
     for (size_t i = 0; i < request.num_headers; i++) {
         auto [it, _] = buf.headers.emplace(request.headers[i].name, request.headers[i].value);
         buf.headers_vector.push_back({ it->first.c_str(), it->second.c_str() });
     }
-    request_copy.headers = buf.headers_vector.data();
 
     auto ud = reinterpret_cast<realm_dart_userdata_async_t>(userdata);
-    ud->scheduler->invoke([ud, request_copy = std::move(request_copy), buf = std::move(buf), request_context]() {
-        (reinterpret_cast<realm_http_request_func_t>(ud->dart_callback)(ud->handle, request_copy, request_context));
+    ud->scheduler->invoke([ud, request = std::move(request), buf = std::move(buf), request_context]() mutable {
+        request.url = buf.url.c_str();
```

Comment:
Please add a comment stating that you are only updating the addresses due to moves, and that all other fields of the request, such as sizes do not need to be modified.

Alternatively, since you know the sizes of everything, You might as well just recreate the whole `request` from scratch rather than moving it around. Your call though.